files/etc/uci-defaults/20-realtek-poe-migrate-poe-type

```diff
@@ -0,0 +1,23 @@
+[ ! -s /etc/config/poe ] && exit 0
+
+. /lib/functions.sh
+
+function migrate_port_poe_type() {
+    local cfg="$1"
+    local poe_plus poe_type
+
+    config_get poe_plus "$cfg" poe_plus ""
+    config_get poe_type "$cfg" poe_type ""
+
+    # Only translate if not already configured
+    [ -z "$poe_type" ] && [ "$poe_plus" = "1" ] \
```

Comment:
I think if statement here would provide better readability:

    if [ "$poe_plus" =  1 ]; then
        do_stuff
    fi

if `poe_plus` is equal to `1`, then obviously, it is not a zero string. So I don't understand the point of the `[ -z ]` part.  Did you mean `if [ -z "$poe_plus" ] || [ "$poe_plus" =  1 ]` ?

Reply:
I went with the style of other uci-defaults files, but I agree an `if [ ... ]; then` is more readable. However, I didn't manage to combine multiple arguments with `-a`. Maybe I was doing something wrong there? Or could this be a busybox limitation?

The check is: "is poe_**type** empty" and "is poe_**plus** set to 1". This is because I don't want to touch sections which already have the new option.

Reply:
Have you tried the following?

    if [ -z "$poe_type" ] && [ "$poe_plus" = 1 ]; then
        do_stuff
    fi